### ISSUE TITLE
Passing Variant to a ByRef explicit type should be allowed in API calls

### ISSUE THREAD

**Issue by camomille22003** [WaynePhillipsEA/twinbasic#175]
**Describe the bug**
cannot change type in a local variant of Function

**To Reproduce**
```
Module ModHelloWorld
    Private Declare Sub GetMem1 Lib "msvbvm60" (ByVal Addr As Long, RetVal As Byte)
    Private Declare Sub GetMem2 Lib "msvbvm60" (ByVal Addr As Long, RetVal As Integer)
    Private Declare Sub GetMem4 Lib "msvbvm60" (ByVal Addr As Long, RetVal As Long)
    Private Declare Sub GetMem8 Lib "msvbvm60" (ByVal Addr As Long, RetVal As Currency)
    Public Sub Main()
        Dim Var_Byte As Byte, Var_Int As Integer, Var_Lng As Long, Var_Curr As Currency
        Var_Byte = 123: Var_Int = 1234: Var_Lng = 123456: Var_Curr = CDec(5234567890#)
        Dim strMsg As String
        strMsg = "Get value of variables by address with PEEK:" & vbCr
        strMsg = strMsg & "BYTE: " & PEEK(VarPtr(Var_Byte), 8) & vbCr
        strMsg = strMsg & "INTEGER: " & PEEK(VarPtr(Var_Int), 16) & vbCr
        strMsg = strMsg & "LONG: " & PEEK(VarPtr(Var_Lng)) & vbCr
        strMsg = strMsg & "CURRENCY: " & PEEK(VarPtr(Var_Curr), 64)
        MsgBox strMsg
    End Sub
    Function PEEK(ByVal Address As Long, Optional ByVal HowMuchBits As Byte = 32) As Variant
    Dim Value As Variant
        Select Case HowMuchBits
            Case 8
                GetMem1 Address, Value
            Case 16
                GetMem2 Address, Value
            Case 32
                GetMem4 Address, Value
            Case 64
                GetMem8 Address, Value
            Case Else
                MsgBox "Invalid value length" & vbCr & vbCr & "Must be one from: 8/16/32/64" & vbCr & vbCr & vbTab & "8 - Byte (unsigned)" & vbCr & vbTab & "16 - Word/Integer" & vbCr & vbTab & "32 - Dword/Long" & vbCr & vbTab & "64 - Qword/Currency"
                Exit Function
        End Select
    PEEK = Value
    End Function
End Module
```

**Expected behavior**
Return Function with correct value as in VB6

**Screenshots**
![Image1](https://user-images.githubusercontent.com/33807881/120436137-db736300-c37e-11eb-9474-6f4a9b56dce1.jpg)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version [e.g. 0.10.810]

**Additional context**
works if Variant is replaced by Byte / Integer / Long or Currency

**Comment by WaynePhillipsEA** [comment]
Interesting one, thanks!   Ordinarily, VB6/VBA does not allow you to pass a Variant to a ByRef parameter of a more explicit type... but weirldy it does allow it for API calls.    Gotta love the classic VB quirks :)

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.1111, thanks for reporting!

**Comment by camomille22003** [comment]
Works perfectly.
Thank you again for these superhuman efforts that have been awaited for 22 years.
